File: common_lib/src/main/java/com/common_lib/base/utils/MD5Utils.java
package com.common_lib.base.utils;

import java.io.UnsupportedEncodingException;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/**
 * Created by alex on 16/7/4.
 */
public class MD5Utils {
    private static MD5Utils instance = null;

    private MD5Utils() {
    }

    public static synchronized MD5Utils getInstance() {
        if (instance == null) {
            instance = new MD5Utils();
        }
        return instance;
    }

    public String getMD5Result(String info){
        try {
            // Create MD5 Hash
            MessageDigest digest = MessageDigest.getInstance("MD5");
            digest.reset();
            digest.update(info.getBytes("UTF-8"));
            byte messageDigest[] = digest.digest();
            // Create Hex String
            StringBuffer hexString = new StringBuffer();
            for (int i=0; i<messageDigest.length; i++){
                if (Integer.toHexString(0xFF & messageDigest[i]).length() == 1)
                    hexString.append("0").append(Integer.toHexString(0xFF & messageDigest[i]));
                else
                    hexString.append(Integer.toHexString(0xFF & messageDigest[i]));
            }
            return hexString.toString();
        } catch (NoSuchAlgorithmException e) {
            e.printStackTrace();
        }catch (UnsupportedEncodingException e){
            e.printStackTrace();
        }
        return null;
    }
}

File: app/src/main/java/com/novel/read/utlis/MD5Utils.kt
package com.novel.read.utlis

import java.security.MessageDigest
import java.security.NoSuchAlgorithmException
import kotlin.experimental.and

/**
 * 将字符串转化为MD5
 */

object MD5Utils {

    private fun strToMd5By32(str: String): String? {
        var reStr: String? = null
        try {
            val md5 = MessageDigest.getInstance("MD5")
            val bytes = md5.digest(str.toByteArray())
            val stringBuffer = StringBuilder()
            for (b in bytes) {
                val bt = b and 0xff.toByte()
                if (bt < 16) {
                    stringBuffer.append(0)
                }
                stringBuffer.append(Integer.toHexString(bt.toInt()))
            }
            reStr = stringBuffer.toString()
        } catch (e: NoSuchAlgorithmException) {
            e.printStackTrace()
        }

        return reStr
    }

    fun strToMd5By16(str: String): String? {
        var reStr = strToMd5By32(str)
        if (reStr != null) {
            reStr = reStr.substring(8, 24)
        }
        return reStr
    }
}
